cmd/etcd-launcher/main.go

```diff
@@ -26,66 +33,137 @@ type envConfig struct {
 	dataDir               string
 	token                 string
 	enableCorruptionCheck bool
+	initialState          string
+}
+
+type etcdCluster struct {
+	config      *envConfig
+	client      *clientv3.Client
+	localClient *clientv3.Client
 }
 
 func main() {
 	// normal workflow, we don't do migrations anymore
-	config, err := getConfigFromEnv()
+	e := &etcdCluster{}
+	err := e.getConfigFromEnv()
 	if err != nil {
 		log.Fatalf("failed to get launcher configuration: %v", err)
 	}
 
+	initialMembers := initialMemberList(e.config.clusterSize, e.config.namespace)
 	// not required, will leave it for now.
-	os.Setenv(initialStateEnvName, "new")
-	os.Setenv(initialClusterEnvName, initialMemberList(config.clusterSize, config.namespace))
+	os.Setenv(initialStateEnvName, e.config.initialState)
+	os.Setenv(initialClusterEnvName, strings.Join(initialMembers, ","))
 
 	log.Print("initializing etcd..")
 	log.Printf("initial-state: %s", os.Getenv(initialStateEnvName))
 	log.Printf("initial-cluster: %s", os.Getenv(initialClusterEnvName))
 
-	err = syscall.Exec(
-		"/usr/local/bin/etcd",
-		etcdCmd(config),
-		os.Environ())
+	// setup and start etcd command
+	cmd := exec.Command("/usr/local/bin/etcd", etcdCmd(e.config)...)
```

Comment:
Shouldn't we make suer that `etcd` cmd exists before doing any calls or preparation? if os.Status("<etcd-path>") => not exist return an error as early as possible?